### PR TITLE
Implement Athena support

### PR DESCRIPTION
**Pull request by arthurnovak**
Please review @motobob @nalundgaard @danilagamma

### CODE REVIEW

File: src/erlcloud_athena.erl [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+batch_get_query_execution(QueryExecutionIds, Config) ->
+    Request = #{<<"QueryExecutionIds">> => QueryExecutionIds},
+    request(Config, "BatchGetQueryExecution", Request).
+
+%% @doc
+%% Athena API:
+%% http://docs.aws.amazon.com/athena/latest/APIReference/API_CreateNamedQuery.html
+%%
+%% `
+%% erlcloud_kinesis:create_named_query(
```
Comment (nalundgaard) [inline src/erlcloud_athena.erl]: s/kinesis/athena?

File: src/erlcloud_athena.erl [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+get_query_execution(QueryExecutionId, Config) ->
+    Request = #{<<"QueryExecutionId">> => QueryExecutionId},
+    request(Config, "GetQueryExecution", Request).
+
+%% @doc
+%% Athena API:
+%% http://docs.aws.amazon.com/athena/latest/APIReference/API_GetQueryResults.html
+%%
+%% `
+%% erlcloud_kinesis:get_query_results(<<"some-query-uuid">>,
```
Comment (nalundgaard) [inline src/erlcloud_athena.erl]: s/kinesis/athena?

**Comment by nalundgaard** [comment]
This doesn't compile on R16. I don't think we have officially retired that in the mainline. Would be nice to make maps optional and only supported on 17/18+.

**Comment by arthurnovak** [comment]
@nalundgaard Do we really want to support R16 yet? 20 has been already released, maybe it's time to get rid of old things to avoid support them.

**Comment by nalundgaard** [comment]
I don't disagree, but it should be formally retired, not inadvertently retired with the introduction of a new module that silently breaks compatibility with no prior warning.

File: src/erlcloud_athena.erl [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+               {"content-type", "application/x-amz-json-1.1"}],
+    Region = erlcloud_aws:aws_region_from_host(Host),
+    erlcloud_aws:sign_v4_headers(Config, Headers, Body, Region, "athena").
+
+get_url(#aws_config{athena_scheme = Scheme,
+                    athena_host   = Host,
+                    athena_port   = Port}) ->
+    Scheme ++ Host ++ ":" ++ integer_to_list(Port).
+
+get_retry_fun() ->
```
Comment (motobob) [inline src/erlcloud_athena.erl]: what is the point of this?

File: src/erlcloud_athena.erl [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+-spec start_query_execution(binary(), binary(), binary(), binary(),
+                            binary() | undefined,
+                            binary() | undefined,
+                            aws_config()) ->
+    {ok, binary()} | {error, any()}.
+start_query_execution(ClientReqToken, Db, Query, OutputLocation,
+                      EncryptionOption, KmsKey, Config) ->
+    EncryptConfig = get_encrypt_config(EncryptionOption, KmsKey),
+    OutputConfig  = #{<<"OutputLocation">> => OutputLocation},
+    ResultConfig  = maps:merge(OutputConfig, EncryptConfig),
```
Comment (motobob) [inline src/erlcloud_athena.erl]: stylish
why not `EncryptConfig#{<<"OutputLocation">> => OutputLocation}`
and same for all other maps:merge()

Comment (arthurnovak) [reply src/erlcloud_athena.erl]: ah gosh, i thought i got rid of this 'merges' everywhere. Just missed those

**Pull request merged**